critical bug

Unsafe source code:

```
use <std>

type number = int | float

def b(a: int) {
    std::println("successfull")
}

def a(n: number){
    b(n) // unsafe downcast to int!
}

def main(){
    var c = 10.0 // float number
    a(c)
}
```

Output:

```
> Task :com.kingmang.ixion.Ixion.main()
Error: Unable to initialize main class test
Caused by: java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    test._a(Ljava/lang/Object;)V @1: invokestatic
  Reason:
    Type 'java/lang/Object' (current frame, stack[0]) is not assignable to integer
  Current Frame:
    bci: @1
    flags: { }
    locals: { 'java/lang/Object' }
    stack: { 'java/lang/Object' }
  Bytecode:
    0000000: 2ab8 001b b1                           

Process finished with exit code 1
```

--- a/src/main/kotlin/com/kingmang/ixion/typechecker/TypeResolver.kt
+++ b/src/main/kotlin/com/kingmang/ixion/typechecker/TypeResolver.kt
@@ -31,7 +31,6 @@ object TypeResolver {
         when {
             par is UnionType && arg is UnionType -> return par.types.containsAll(arg.types) && arg.types.containsAll(par.types)
             par is UnionType && par.types.contains(arg) -> return true
-            arg is UnionType && arg.types.contains(par) -> return true
         }
 
         // any

